src/engine/threaded_engine.h

```diff
@@ -177,7 +177,11 @@ class ThreadedVar final
   static std::atomic<std::size_t> counter;
   ~ThreadedVar() { LOG(INFO) << __func__ << " " << --counter; }
 #endif  // ENGINE_DEBUG
-  /*! \brief exception_ptr associated with the ThreadedVar */
+  /*!
+   * \brief exception_ptr associated with the ThreadedOpr
+   * cannot modify state of exception object since dereferencing
+   * exception_ptr is undefined behavior. Using shared_ptr to hold
```

Comment:
Look likes [exception_ptr](https://en.cppreference.com/w/cpp/error/exception_ptr) is already a shared-ownership smart pointer. Would that be sufficient for our use cases, instead of wrapping it with shared_ptr?

Reply:
the purpose is to update the object that exception ptr points to and dereferencing an exception_ptr is undefined behavior. So wrapping it in shared_ptr is another way.